packngo.go

```diff
@@ -355,6 +355,13 @@ func NewClientWithAuth(consumerToken string, apiKey string, httpClient *http.Cli
 // NewClientWithBaseURL returns a Client pointing to nonstandard API URL, e.g.
 // for mocking the remote API
 func NewClientWithBaseURL(consumerToken string, apiKey string, httpClient *http.Client, apiBaseURL string) (*Client, error) {
+	return NewClientWithHeader(consumerToken, apiKey, httpClient, apiBaseURL, nil)
+}
+
+// NewClientWithHeader returns a Client which for every new request adds given HTTP header set. Otherwise it behaves
+// like NewClientWithBaseURL. The headers X-Auth-Token, X-Consumer-Token, User-Agent will be ignored if provided
+// in the set.
+func NewClientWithHeader(consumerToken string, apiKey string, httpClient *http.Client, apiBaseURL string, header http.Header) (*Client, error) {
```

Comment:
I've considered adding other constructors - adding a new one for each new field we add later may be messy.

What are your thoughts on changing `NewClient()` to `NewClient(...ClientOpt)` where `ClientOpt` is `func (*Client) error`.  We can then add `ClientOpt` helpers for headers, base URL, Auth, UserAgent, etc.

Reply:
Hi @displague , do you mean - as part of this PR?
I'm familiar with the convention and I can add it, my only concern is that's not super-discoverable for users. (A comment would fix that.)

The other fancy option are chained builders:

```
c := NewClient()
```

```
c := NewClient().WithHeaders(h)
```

```
c := NewClient().
        WithHeaders().
        WithSomethingElse().
        WithYetAnotherThingie()
```

What do you think?

Reply:
It's good that you're considering discoverability.

This particular pattern won't against NewClient which returns `(*Client,error)` today.

Reply:
Added :smile: